buildSrc/src/main/java/org/opensearch/gradle/VersionProperties.java

```diff
@@ -101,10 +101,11 @@ public static String getBundledJdkVendor() {
 
     private static Properties getVersionProperties() {
         Properties props = new Properties();
-        try (InputStream propsStream = VersionProperties.class.getResourceAsStream("/version.properties")) {
-            if (propsStream == null) {
-                throw new IllegalStateException("/version.properties resource missing");
-            }
+        InputStream propsStream = VersionProperties.class.getResourceAsStream("/version.properties");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *propsStream*. Currently, there are execution paths that do not contain closure statements, for example, when Properties.load throws an exception. Ensure *propsStream* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).